packages/Telephony/Actions/CallTransfer.cs

```diff
@@ -55,12 +56,24 @@ public CallTransfer([CallerFilePath] string sourceFilePath = "", [CallerLineNumb
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         public async override Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var phoneNumber = this.PhoneNumber?.GetValue(dc.State);
+            if (dc.Context.Activity.ChannelId == Channels.Telephony)
+            {
+                var error = "Phone number is empty, please provide number in E.164 format";
+                var regexFormat = "^\\+\\d{1,3}\\s\\d{1,14}(\\s\\d{1,13})?";
 
-            // Create handoff event, passing the phone number to transfer to as context.
-            HandoffContext = new { TargetPhoneNumber = phoneNumber };
-            
-            return await base.BeginDialogAsync(dc, options, cancellationToken).ConfigureAwait(false);
+                var phoneNumber = this.PhoneNumber?.GetValue(dc.State);
+
+                if (string.IsNullOrEmpty(phoneNumber) || Regex.IsMatch(phoneNumber, regexFormat))
```

Comment:
Do you mean &&

Reply:
Bit more detail: This condition does not work as-is, since if the regex is a match (i.e. good phone number) we'll send an error and never transfer the call. Something like `if (string.IsNullOrEmpty(phoneNumber) && !Regex.IsMatch(phoneNumber, regexFormat))` could yield the desired results. However again, let's discuss whether we want or not to include the validation on Tuesday.

You might wonder, if you did run the tests, why did they pass? The reason your test could have passed if you ran them is that 1) the regex is wrong , so it will never return true, and 2) the tests you added are all for non-telephony channel (which you specified in the cs explicitly ) so the behavior will not exercise this code

Reply:
removing this validation